src/operator/numpy/random/np_bernoulli_op.cc

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2019 by Contributors
+ * \file np_bernoulli_op.cc
+ * \brief Operator for numpy sampling from bernoulli distributions
+ */
+
+#include "./np_bernoulli_op.h"
+#include "./dist_common.h"
+
+namespace mxnet {
+namespace op {
+
+DMLC_REGISTER_PARAMETER(NumpyBernoulliParam);
+
+NNVM_REGISTER_OP(_npi_bernoulli)
+.describe("Sample frmo bernoulli distribution")
+.set_num_inputs(
+  [](const nnvm::NodeAttrs& attrs) {
+    const NumpyBernoulliParam& param = nnvm::get<NumpyBernoulliParam>(attrs.parsed);
+    int num_inputs = 1;
+    if (param.logit.has_value() || param.prob.has_value()) {
+      num_inputs -= 1;
+    }
+    return num_inputs;
+  }
+)
+.set_num_outputs(1)
+.set_attr<nnvm::FListInputNames>("FListInputNames",
+  [](const NodeAttrs& attrs) {
+    const NumpyBernoulliParam& param = nnvm::get<NumpyBernoulliParam>(attrs.parsed);
+    int num_inputs = 1;
+    if (param.logit.has_value() || param.prob.has_value()) {
+      num_inputs -= 1;
+    }
+    if (num_inputs == 0) return std::vector<std::string>();
+    return std::vector<std::string>{"input1"};
```

Comment:
We can have a single if-else to return the value.

Reply:
@sxjscience You mean something like
```c++
return (num_inputs == 0) ? std::vector<std::string>() : std::vector<std::string>{"input1"};
```
?

Reply:
@sxjscience  Using one if-else would cause lint error if my memory serves.
@haojin2 's solution is a feasible one, I will switch the ternary expression form.

Reply:
I mean something like “ (param.logit.has_value() || param.prob.has_value()) ? A : B” or use an if-else